examples/step-15/doc/results.dox

```diff
@@ -251,3 +251,51 @@ done implicitly? That is in fact possible, and runs under the name
 "automatic differentiation". step-71 discusses this very
 concept in general terms, and step-72 illustrates how this can be
 applied in practice for the very problem we are considering here.
+
+
+<h4> Storing the Jacobian matrix in lower-precision floating point variables </h4>
+
+On modern computer systems, *accessing* data in main memory takes far
+longer than *actually doing* something with it: We can do many floating
+point operations for the time it takes to load one floating point
+number from memory onto the processor. Unfortunately, when we do things
+such as matrix-vector products, we only multiply each matrix entry once
+with another number (the corresponding entry of the vector) and then we
+add it to something else -- so two floating point operations for one
+load. (Strictly speaking, we also have to load the corresponding vector
+entry, but at least sometimes we get to re-use that vector entry in
+doing the products that correspond to the next row of the matrix.) This
+is a fairly low "arithmetic intensity", and consequently we spend most
+of our time during matrix-vector products waiting for data to arrive
+from memory rather than actually doing floating point operations.
+
+This is of course one of the rationales for the "matrix-free" approach to
+solving linear systems (see step-37, for example). But if you don't quite
+want to go all that way to change the structure of the program, then
+here is a different approach: Storing the system matrix (the "Jacobian")
+in single-precision instead of double precision floating point numbers
+(i.e., using `float` instead of `double` as the data type). This reduces
+the amount of memory necessary by a factor of two, and consequently
```

Comment:
This is not entirely correct, because the column indices in the matrix take the same space (4 bytes per entry) irrespective of the number type (4 or 8 bytes per entry). Thus, the saving is around a factor of 1.3-1.5. I agree we should not make the discussion here more complicated than necessary, so I suggest to add a short note towards the end of this paragraph that explains that a sparse matrix storage also consists of other terms. An interesting follow-up question would be to which extent one could compress the integer part of a sparse matrix more. There have been some papers in 2008-2009, but I never wanted to implement those because we would need them in Trilinos/PETSc and not in deal.II, and then matrix-free methods give even bigger gains.